escrow/etcd_storage.go

```diff
@@ -0,0 +1,121 @@
+package escrow
+
+import (
+	"bytes"
+	"encoding/gob"
+
+	"github.com/singnet/snet-daemon/etcddb"
+	"github.com/spf13/viper"
+)
+
+// EtcdStorage storage based on etcd
+type EtcdStorage struct {
+	client *etcddb.EtcdClient
+}
+
+// NewEtcdStorage create an etcd storage from Ciper config
+func NewEtcdStorage(vip *viper.Viper) (storage *EtcdStorage, err error) {
+	client, err := etcddb.NewEtcdClient(vip)
+
+	if err != nil {
+		return
+	}
+
+	storage = &EtcdStorage{client: client}
+	return
+}
+
+// Get gets value from etcd by key
+func (storage *EtcdStorage) Get(key *PaymentChannelKey) (state *PaymentChannelData, ok bool, err error) {
+
+	keyBytes, err := serialize(key)
+	if err != nil {
+		return
+	}
+
+	valueBytes, err := storage.client.Get(keyBytes)
+
+	if err != nil || len(valueBytes) == 0 {
```

Comment:
How do the caller know whether value is found or not? Get() method which is implemented here has ok flag which is false if key was not found. As far as I see EtcdClient.Get method returns nil when key is not found. Why len(valueBytes) is checked here?

Reply:
I added the ok to the etcd client Get return value in the branch:
https://github.com/stellarspot/snet-daemon/commit/c417f27d50de9fa58cc14daf083cc33484455a34